src/typechecker/environment.lisp

```diff
@@ -576,19 +570,16 @@
   (predicate           (util:required 'predicate)           :type ty-predicate        :read-only t)
   (superclasses        (util:required 'superclasses)        :type ty-predicate-list   :read-only t)
   (class-variables     (util:required 'class-variables)     :type util:symbol-list    :read-only t)
-
-  ;; Hash table mapping variable symbols to their index in the predicate
-  (class-variable-map  (util:required 'class-variable-map)  :type environment-map     :read-only t)
   (fundeps             (util:required 'fundeps)             :type fundep-list         :read-only t)
 
   ;; Methods of the class containing the same tyvars in PREDICATE for
   ;; use in pretty printing
   (unqualified-methods (util:required 'unqualified-methods) :type ty-class-method-list :read-only t)
   (codegen-sym         (util:required 'codegen-sym)         :type symbol               :read-only t)
   (superclass-dict     (util:required 'superclass-dict)     :type list                 :read-only t)
-  (superclass-map      (util:required 'superclass-map)      :type environment-map      :read-only t)
+  (superclass-map      (util:required 'superclass-map)      :type list                 :read-only t)
```

Comment:
Can you write a comment describing the keys and values here? I don't think "list" is self-evident. Something like

```
;; an immutable alist of the form ((a1 b1) ...) whose keys are ... and values are ...
```

clarify if it's an alist of lists or an alist of conses.

I would say this is the sole and biggest negative of this change, we are using an ad hoc data structure.